source/custom/nimxlcterminaladaptor_restricted_converters.h

```diff
@@ -31,20 +29,39 @@ struct NIErrStatusOutputConverter {
     return &status;
   }
 
+  std::string UrlEncode(char* input) const
+  {
+    std::string encoded(input);
+    // Replace the following characters to their equivalent '%xx' hex code
+    const char* charsToEncode = "!*'();:@&=+$,/?%#[] <>~";
+    for (size_t position = 0;
+          ((position = encoded.find_first_of(charsToEncode, position)) != std::string::npos);
+          // If an unsafe character is found, advance past the replacement hex code
+          position += 3) {
+        char character[4];
+        ::_snprintf(character, 4, "%%%2x", static_cast<int>(encoded[position]));
+        encoded.replace(position, 1, character);
+    }
+    return encoded;
+  }
+
   void to_grpc(nimxlcterminaladaptor_restricted_grpc::NIErrStatus& output) const
   {
     output.set_code(status.code);
-    output.set_capacity(status.capacity);
-    if (status.json != nullptr)
+    if (nierr_Status_isFatal(&status))
     {
-      output.set_json(status.json);
+      std::string description = "Error " + std::to_string(status.code) + " has occurred in nimxlcTerminalAdaptor. Refer to the trailing metadata for details.";
```

Comment:
This description belongs in the `grpc::Status` object. 

If you want non-error cases to return a `NIErrStatus` protobuf message to support warnings, that's fine, but it should contain `code` and `json`.

Reply:
This is not fixed.

Reply:
This should be fixed now.